epoxy-composeinterop-maverickssample/src/main/java/com/airbnb/epoxy/composeinterop/maverickssample/MultiKeyComposeInteropFragment.kt

```diff
@@ -0,0 +1,325 @@
+package com.airbnb.epoxy.composeinterop.maverickssample
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.material.Divider
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.unit.dp
+import androidx.fragment.app.Fragment
+import com.airbnb.epoxy.EpoxyRecyclerView
+import com.airbnb.epoxy.ModelCollector
+import com.airbnb.epoxy.TypedEpoxyController
+import com.airbnb.epoxy.composableInterop
+import com.airbnb.mvrx.MavericksState
+import com.airbnb.mvrx.MavericksView
+import com.airbnb.mvrx.MavericksViewModel
+import com.airbnb.mvrx.fragmentViewModel
+import com.airbnb.mvrx.withState
+
+data class MultiKeyComposeInteropState(
+    val keyAsInt: Int = 0,
+    val keyAsString: String = "",
+    val keyAsList: List<Int> = emptyList(),
+    val keyAsDataClass: DataClassKey = DataClassKey()
+) : MavericksState
+
+data class DataClassKey(
+    var listInDataClass: MutableList<Int> = mutableListOf()
+)
+
+class MultiKeyComposeInteropViewModel(initialState: MultiKeyComposeInteropState) :
+    MavericksViewModel<MultiKeyComposeInteropState>(initialState) {
+
+    fun increaseCounter() {
+        withState { state ->
+            val updatedCounter = state.keyAsInt + 1
+            setState { copy(keyAsInt = updatedCounter) }
+        }
+    }
+
+    fun appendKeyString() {
+        withState { state ->
+            val updatedString = state.keyAsString + "#"
+            setState { copy(keyAsString = updatedString) }
+        }
+    }
+
+    fun appendKeyList() {
+        withState { state ->
+            val updatedList = state.keyAsList.toMutableList()
+            updatedList.add(1)
+            setState { copy(keyAsList = updatedList) }
+        }
+    }
+
+    fun appendDataClass() {
+        withState { state ->
+            val updatedDataClass =
+                state.keyAsDataClass.copy(listInDataClass = state.keyAsDataClass.listInDataClass.toMutableList())
+
+            updatedDataClass.listInDataClass.add(2)
+
+            setState { copy(keyAsDataClass = updatedDataClass) }
+        }
+    }
+
+    val mutableCounter = mutableStateOf(0)
```

Comment:
optional: this could very well be a flow or a live data and could be converted into a Compose compatible state object using the `collectAs/observeAs` extension methods. This allows for clean separation between compose and the rest of our architecture.